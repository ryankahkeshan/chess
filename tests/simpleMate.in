setup
+ K a1 + k e8
+ r g2
+ q h3
done game human human

move a1 a2
move a1 b1
move h3 h1
