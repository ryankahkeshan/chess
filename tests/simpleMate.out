8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 K _ _ _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 K _ _ _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _r_
1 K _ _ _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _q
2  _ _ _r_
1 K _ _ _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _q
2  _ _ _r_
1 K _ _ _ 

  abcdefgh

Invalid move
8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _q
2  _ _ _r_
1 _K_ _ _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _r_
1 _K_ _ _q

  abcdefgh

Checkmate! Black wins!
Final Score:
White: 0
Black: 1
